### PR TITLE
gh-99051: remove duplicated test from `test_weakref`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-99051

### CODE REVIEW

**Review by Jason-Y-Z — APPROVED** [review]
Given that all Python 3 classes inherit from `object` by default, this looks good.
A slight comment would be maybe we want to update the test names since we now jump from `test_callback_in_cycle_1` to `test_callback_in_cycle_3`.

**Comment by iritkatriel** [comment]
Note that this comment in test_callback_in_cycle_3 becomes obsolete if we merge this as it is now:

"# This one broke the first patch that fixed the last two."

Can you think of more meaningful suffixes for these test names than a number?

Also, maybe remove the `(object)` in the class definitions in the other tests while we're clearing up old-style classes?

**Comment by sobolevn** [comment]
> Can you think of more meaningful suffixes for these test names than a number?

Yes, this is a good suggestion. Thank you.
I've used more semantic names.

> Also, maybe remove the (object) in the class definitions in the other tests while we're clearing up old-style classes?

I would prefer to keep it as is, because there's not semantical difference.
And style-only diffs are not something I would like to see in a `git blame` :)

File: Lib/test/test_weakref.py [hunk 1]
```diff
@@ -637,11 +637,11 @@ def acallback(self, ignore):
         del I, J, II
         gc.collect()
 
-    def test_callback_in_cycle_3(self):
+    def test_callback_reachable_one_way(self):
         import gc
 
-        # This one broke the first patch that fixed the last two.  In this
-        # case, the objects reachable from the callback aren't also reachable
+        # In this case,
```
Comment (iritkatriel) [inline Lib/test/test_weakref.py]:
```suggestion
        # This one broke the first patch that fixed the previous test. In this case,
```

**Review by iritkatriel — APPROVED** [review]
Looks good, just one nit about the comment.

**Comment by sobolevn** [comment]
Thanks again! 👍

**Pull request merged**